### DISCUSSION TITLE
interactive visualization

### DISCUSSION THREAD

**Discussion by decareano** [theOehrly/Fast-F1#344] [General]
Hi, I am beginning to use fast f1 and matplotlib for visualizations. How can I get the graphics to render so that I can hover over the graphic with a vertical line to see the different values(ie: speed, brake, throttle, etc). F1 Tempo does that but it's kind of limited in what you can render. Thanks

**Comment by Casper-Guo** [comment]
Matplotlib itself has very poor support for interactivity. You might want to look into [Plotly ](https://plotly.com/python/) instead for that purpose

**Reply by theOehrly** [reply]
For an easy solution, I agree with @Casper-Guo. Plotly is probably the way to go. 

It can be done in matplotlib, but you will have to write the code for the animated overlays yourself. That would be a more complicated endeavour.

**Comment by decareano** [comment]
Thank you. I will look into Plotly.

**Reply by decareano** [reply]
Hi, the info on Plotly is so big. can you guys point me in the direction of a tutorial or some racing code samples where I can see Plotly in action? thanks.

**Reply by andrewestes41** [reply]
This is pretty basic, but it shows plotly in action

https://plotly.com/python/line-charts/

**Comment by decareano** [comment]
Hi, I was wondering if there is an API for IndyCars in the US?. I live in the US and in addition to F1 I follow the IndyCar series. thanks.

**Reply by theOehrly** [reply]
There is, yes. Support for other series is something that will be potentially added at some point. But absolutely no guarantees for this. Might be this year, next year, or never.

**Comment by decareano** [comment]
That would be very good. Do you know of an existing one? I checked Sportradar USA API but they don't have features like brakes, throttle, RPM, gear, etc.

**Reply by theOehrly** [reply]
Apparently you can get the data from Indycar servers directly. But I don't think that's a documented API. I've not really looked into it, though. There are things with higher priority at the moment.

**Comment by decareano** [comment]
Thanks. I see some data from IndyCar in text form. But you are right, I don't see a documented API. If you ever tackle the issue, let me know. I have some programming experience (mostly java, some python) and a bit of API configuration.

**Comment by decareano** [comment]
Hi, is there an API call to obtain information on tyres, say, during the race (ie: lap 30 car #1 is on Softs). Thanks.

**Reply by theOehrly** [reply]
The ``Laps`` DataFrame has a ``Compound`` column where this information is available.

**Reply by decareano** [reply]
Thank you. I found it. !!
